staging/src/k8s.io/apiserver/pkg/storage/etcd/api_object_versioner.go

```diff
@@ -25,7 +25,8 @@ import (
 )
 
 // APIObjectVersioner implements versioning and extracting etcd node information
-// for objects that have an embedded ObjectMeta or ListMeta field.
+// for objects that have an embedded ObjectMeta or ListMeta field. It will clear
+// the self link when the object version is empty.
```

Comment:
Why "when the object version is empty"?
I guess I'm missing something, but I'm not really sure what...